elements/pl-units-input/pl-units-input.py

```diff
@@ -0,0 +1,467 @@
+            "show_info": show_info,
+            "placeholder": placeholder_text,
+            "uuid": pl.get_uuid(),
+            display.value: True,
+            "display_append_span": show_info or suffix,
+        }
+
+        if score is not None:
+            score_type, score_value = pl.determine_score_params(score)
+            html_params[score_type] = score_value
+
+        if raw_submitted_answer is not None:
+            html_params["raw_submitted_answer"] = escape(raw_submitted_answer)
+
+        with open(UNITS_INPUT_MUSTACHE_TEMPLATE_NAME, "r", encoding="utf-8") as f:
+            return chevron.render(f, html_params).strip()
+
+    elif data["panel"] == "submission":
+        parse_error = data["format_errors"].get(name, None)
+        html_params = {
+            "submission": True,
+            "label": label,
+            "parse_error": parse_error,
+            "uuid": pl.get_uuid(),
+        }
+
+        if parse_error is None and name in data["submitted_answers"]:
+            a_sub = data["submitted_answers"].get(name, None)
+            if a_sub is None:
+                raise ValueError("submitted answer is None")
+
+            html_params["suffix"] = suffix
+            html_params["a_sub"] = a_sub
+            html_params["raw_submitted_answer"] = raw_submitted_answer
+
+        elif name not in data["submitted_answers"]:
+            html_params["missing_input"] = True
+            html_params["parse_error"] = None
+
+        else:
+            submitted_answer = data["raw_submitted_answers"].get(name, None)
+            if submitted_answer is not None:
+                html_params["raw_submitted_answer"] = pl.escape_unicode_string(
+                    submitted_answer
+                )
+
+        if score is not None:
+            score_type, score_value = pl.determine_score_params(score)
+            html_params[score_type] = score_value
+
+        feedback = partial_scores.get("feedback")
+        allow_feedback = pl.get_boolean_attrib(
+            element, "allow-feedback", ALLOW_FEEDBACK_DEFAULT
+        )
+        if feedback is not None and allow_feedback:
+            html_params["feedback"] = feedback
+
+        html_params["error"] = html_params["parse_error"] or html_params.get(
+            "missing_input", False
+        )
+        with open(UNITS_INPUT_MUSTACHE_TEMPLATE_NAME, "r", encoding="utf-8") as f:
+            return chevron.render(f, html_params).strip()
+
+    elif data["panel"] == "answer":
+        a_tru = data["correct_answers"].get(name, None)
+
+        if a_tru is None:
+            return ""
+
+        ureg = pl.get_unit_registry()
+        a_tru_parsed = ureg.Quantity(a_tru)
+
+        if grading_mode is GradingMode.ONLY_UNITS:
+            a_tru_string = f"{a_tru_parsed.units}"
+        else:
+            # Display reference solution with the given number of digits
+            # TODO double check if this is confusing or not
+            a_tru_string = f"{a_tru_parsed:.{digits}g}"
```

Comment:
Note: should change this to just take in a custom format string instead of using digits at all.